Extensions/PrimitiveDrawing/Extension.cpp

```diff
@@ -168,9 +187,194 @@ void DeclarePrimitiveDrawingExtension(gd::PlatformExtension& extension) {
       .AddParameter("expression", _("X coordinate end arc"))
       .AddParameter("expression", _("Y coordinate end arc"))
       .AddParameter("expression", _("Radius (in pixels)"))
+      .AddParameter("yesorno", _("Close path"))
       .SetFunctionName("DrawArcTo")
       .SetIncludeFile("PrimitiveDrawing/ShapePainterObject.h");
 */
+  obj.AddAction("BezierCurve",
+                _("Bezier curve"),
+                _("Draw a bezier curve on screen"),
+                _("Draw from _PARAM1_;_PARAM2_ to _PARAM7_;_PARAM5_ a bezier curve (first control point: _PARAM3_;_PARAM4_, second control point: _PARAM5_;_PARAM6_) "
+                  " with _PARAM0_"),
+                _("Drawing"),
+                "res/actions/delete24.png",
+                "res/actions/delete.png")
+
+      .AddParameter("object", _("Shape Painter object"), "Drawer")
+      .AddParameter("expression", _("X position of start point"))
+      .AddParameter("expression", _("Y position of start point"))
+      .AddParameter("expression", _("First control point x"))
+      .AddParameter("expression", _("First control point y"))
+      .AddParameter("expression", _("Second Control point x"))
+      .AddParameter("expression", _("Second Control point y"))
+      .AddParameter("expression", _("Destination point x"))
+      .AddParameter("expression", _("Destination point y"))
+      .SetFunctionName("drawBezierCurve")
+      .SetIncludeFile("PrimitiveDrawing/ShapePainterObject.h");
+
+  obj.AddAction("QuadraticCurve",
+                _("Quadratic curve"),
+                _("Draw a quadratic curve on screen"),
+                _("Draw from _PARAM1_;_PARAM2_ to _PARAM5_;_PARAM6_ a quadratic curve (control point: _PARAM3_;_PARAM4_) "
+                "with _PARAM0_"),
+                _("Drawing"),
+                "res/actions/delete24.png",
+                "res/actions/delete.png")
+
+      .AddParameter("object", _("Shape Painter object"), "Drawer")
+      .AddParameter("expression", _("X position of start point"))
+      .AddParameter("expression", _("Y position of start point"))
+      .AddParameter("expression", _("Control point x"))
+      .AddParameter("expression", _("Control point y"))
+      .AddParameter("expression", _("Destination point x"))
+      .AddParameter("expression", _("Destination point y"))
+      .SetFunctionName("drawQuadraticCurve")
+      .SetIncludeFile("PrimitiveDrawing/ShapePainterObject.h");
+
+  obj.AddAction("BeginFillPath",
+                _("Begin fill path"),
+                _("Begins drawing filling of an advanced path"),
```

Comment:
Begin to draw a simple one-color fill. Subsequent actions, such as "Path line" (in the Advanced category) can be used to draw. Be sure to use "End fill path" action when you're done drawing the shape.